### PR TITLE
[LOG-1257][Housekeeping] Avoid malformed json

### PR DESCRIPTION
**Pull request by Diegores14**
### Summary ###
The shopper App sometimes sends malformed JSONs. At the moment they are not able to replicate the problem. when these corrupt messages arrive, it causes the position update to not behave the way we want, for this reason we decided to avoid these messages.

### Test ####
- Test in local environment.

### CODE REVIEW

**Review by osdajigu — DISMISSED** [review]

File: src/rabbitmq/index.ts [hunk 1]
```diff
@@ -170,32 +170,37 @@ const wrapper = (configName: string): RabbitMQ => {
...
             try {
-              await channelConfig.processor(messages, lastMessage)
-              channel.ack(lastMessage, true)
+              messages[index] = JSON.parse(value.content.toString())
             } catch (err) {
-              logger.error('processing_failed', { messages, lastMessage }, err)
-              channel.nack(lastMessage, true)
+              messageToDelete.push(value)
+              logger.error('parsing_failed', { message: value.content.toString() }, err)
+              channel.nack(value, false, false)
```
Comment (joagranadosme) [inline src/rabbitmq/index.ts:193]: Why is this necessary if you are acknowledging the last message always?

File: src/rabbitmq/index.ts [hunk 2]
```diff
@@ -170,32 +170,37 @@ const wrapper = (configName: string): RabbitMQ => {
...
         if (lastMessage) {
-          try {
-            messages.forEach((value, index) => {
-              messages[index] = JSON.parse(value)
-            })
+          messages.forEach((value, index) => {
             try {
-              await channelConfig.processor(messages, lastMessage)
-              channel.ack(lastMessage, true)
+              messages[index] = JSON.parse(value.content.toString())
```
Comment (joagranadosme) [inline src/rabbitmq/index.ts:189]: This validation can be done inside the while loop instead of doing it twice and after that a filter.

File: src/rabbitmq/index.ts (hunk 1 above)
Comment (Diegores14) [reply src/rabbitmq/index.ts:193]: This sends the bad message to the error queue. and thus we avoid re-queuing to the original queue

Comment (joagranadosme) [reply src/rabbitmq/index.ts]: Are you sure this can be done like that? If you ack the last that means that all messages will be processed. You don't have to send that message to the errors queue.

Comment (Diegores14) [reply src/rabbitmq/index.ts]: yes, I test that, if a message arrives with the malformed json, this message is sent to the error queue and the rest are processed ordinary. and the bad message never goes back to the queue the consumer is subscribed to.

**Review by joagranadosme — APPROVED** [review]

**Review by cerodriguezl — APPROVED** [review]
